skunkworks_crow/src/main/java/org/odk/share/activities/ScannerActivity.java

```diff
@@ -0,0 +1,133 @@
+package org.odk.share.activities;
+
+/*
+ * created by Chromicle
+ */
+
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.view.KeyEvent;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.Button;
+
+import com.journeyapps.barcodescanner.CaptureManager;
+import com.journeyapps.barcodescanner.DecoratedBarcodeView;
+
+import org.odk.share.R;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+
+public class ScannerActivity extends AppCompatActivity implements
+        DecoratedBarcodeView.TorchListener {
+
+    private CaptureManager capture;
+    private DecoratedBarcodeView decoratedScannerView;
+    private Button switchFlashlightButton;
+    private boolean isFlashLightOn = false;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_scanner);
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
+
+
+        //Initialize barcode scanner view
+
+        decoratedScannerView = findViewById(R.id.zxing_barcode_scanner);
```

Comment:
Please replace this with Butterknife annotation

Reply:
This one is still left

Reply:
if I am using butterknife then it is producing null pointer exception and app is going to crash can you help me on that

Reply:
That might happen if you are not binding views using `Butterknife.bind(this)` before using any injected variable. 
In this case, you might be trying to `setTorchListener`. So, the solution is to do the binding first